lib/src/main/java/com/auth0/jwt/JWTVerifier.java

```diff
@@ -316,17 +325,20 @@ private void assertNonNull(String name) {
             }
         }
 
-        private void addLeewayToDateClaims() {
-            if (!claims.containsKey(PublicClaims.EXPIRES_AT)) {
-                claims.put(PublicClaims.EXPIRES_AT, defaultLeeway);
-            }
-            if (!claims.containsKey(PublicClaims.NOT_BEFORE)) {
-                claims.put(PublicClaims.NOT_BEFORE, defaultLeeway);
-            }
-            if (!claims.containsKey(PublicClaims.ISSUED_AT)) {
-                claims.put(PublicClaims.ISSUED_AT, defaultLeeway);
-            }
-        }
+		private void addLeewayToDateClaims() {
+			if (!claims.containsKey(PublicClaims.EXPIRES_AT)) {
+				claims.put(PublicClaims.EXPIRES_AT, defaultLeeway);
+			}
+			if (!claims.containsKey(PublicClaims.NOT_BEFORE)) {
+				claims.put(PublicClaims.NOT_BEFORE, defaultLeeway);
+			}
+			if (!claims.containsKey(PublicClaims.ISSUED_AT)) {
+				claims.put(PublicClaims.ISSUED_AT, defaultLeeway);
+			}
+			if(ignoreIssuedAt) {
```

Comment:
swap this if with the one above so you avoid the case in which the claim is added and immediately removed. You could also `return` right after removing the claim, so the following if is not considered.

Reply:
done, thank you.